CONFIGURATION.md

```diff
@@ -96,7 +96,7 @@ interceptors                             |  *  |                 |
 group.id                                 |  C  |                 |               | high       | Client group id string. All clients sharing the same group.id belong to the same group. <br>*Type: string*
 group.instance.id                        |  C  |                 |               | medium     | Enable static group membership. Static group members are able to leave and rejoin a group within the configured `session.timeout.ms` without prompting a group rebalance. This should be used in combination with a larger `session.timeout.ms` to avoid group rebalances caused by transient unavailability (e.g. process restarts). Requires broker version >= 2.3.0. <br>*Type: string*
 partition.assignment.strategy            |  C  |                 | range,roundrobin | medium     | The name of one or more partition assignment strategies. The elected group leader will use a strategy supported by all members of the group to assign partitions to group members. If there is more than one eligible strategy, preference is determined by the order of this list (strategies earlier in the list have higher priority). Cooperative and non-cooperative (eager) strategies must not be mixed. Available strategies: range, roundrobin, cooperative-sticky. <br>*Type: string*
-session.timeout.ms                       |  C  | 1 .. 3600000    |         10000 | high       | Client group session and failure detection timeout. The consumer sends periodic heartbeats (heartbeat.interval.ms) to indicate its liveness to the broker. If no hearts are received by the broker for a group member within the session timeout, the broker will remove the consumer from the group and trigger a rebalance. The allowed range is configured with the **broker** configuration properties `group.min.session.timeout.ms` and `group.max.session.timeout.ms`. Also see `max.poll.interval.ms`. <br>*Type: integer*
+session.timeout.ms                       |  C  | 1 .. 3600000    |         45000 | high       | Client group session and failure detection timeout. The consumer sends periodic heartbeats (heartbeat.interval.ms) to indicate its liveness to the broker. If no hearts are received by the broker for a group member within the session timeout, the broker will remove the consumer from the group and trigger a rebalance. The allowed range is configured with the **broker** configuration properties `group.min.session.timeout.ms` and `group.max.session.timeout.ms`. Also see `max.poll.interval.ms`. <br>*Type: integer*
```

Comment:
The kip talks about the interplay between `session.timeout.ms` and `request.timeout.ms` (30s). librdkafka's `socket.timeout.ms` defaults to 60s, which is still higher than 45s. Do similar considerations apply here?

Side note: I don't like the name `socket.timeout.ms`, because it sounds like a lower level thing than it is. Have you considered adding an alias `request.timeout.ms`?

Reply:
Very sharp eyed!

All the group-related requests have their own timeouts, namely:
OffsetCommit = 60s (socket.timeout.ms)
JoinGroup:
                /* Request timeout is max.poll.interval.ms + grace
                 * if the broker supports it, else
                 * session.timeout.ms + grace. */
                (ApiVersion >= 1 ?
                 rk->rk_conf.max_poll_interval_ms :
                 rk->rk_conf.group_session_timeout_ms) +
                3000/* 3s grace period*/,
SyncGroup: session.timeout.ms + 3s
LeaveGroup: 5s (hardcoded)


As for socket.timeout.ms - it is from the Scala days, and unfortunately we already have request.timeout.ms which controls the ProduceRequest.RequestTimeoutMs field which is the broker-side timeout for produce requests.

Reply:
it may be worth considering `api.request.timeout.ms`, though it's clunky. i guess probably better to just leave it as you say.